File: crossflow/org.eclipse.scava.crossflow.tests/src/org/eclipse/scava/crossflow/tests/wordcount/LineSource.java
package org.eclipse.scava.crossflow.tests.wordcount;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

public class LineSource extends LineSourceBase {
	
	@Override
	public void produce() throws Exception {
		for (File file : workflow.getInputDirectory().listFiles()) {
			try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
				String line = null;
				while ((line = reader.readLine()) != null) {
					sendToLines(new Line(line));
				}
			}
		}
	}

}

File: crossflow/org.eclipse.scava.crossflow.examples.wordcount/src-py/org/eclipse/scava/crossflow/examples/wordcount/LineSource.py
import threading
from org.eclipse.scava.crossflow.examples.wordcount.LineSourceBase import LineSourceBase
from org.eclipse.scava.crossflow.examples.wordcount.Line import Line

class LineSource(LineSourceBase):

	def __init__(self):
		super().__init__()
	
	def produce(self):
		def run():
			# TODO: Add implementation that instantiates, sets, and submits source objects (example below)
			line1 = Line()
			#	line1.setText( String )
		
			self.sendToLines( line1)
			
			threading.Timer(100, run).start()
		run()			
